src/Service/Scheduler.php

```diff
@@ -7,6 +7,72 @@
 
 class Scheduler {
 
+	protected string $jobhook; 
+	protected string $reccurence; 
+	protected int $timestamp;
+
+	const UNSCHEDULER_HOOK = COMMONSBOOKING_PLUGIN_SLUG . '_unschedule';
+
+	//constructs the class, if job does not exist yet it is created
+	function __construct(
+		string $jobhook, //the action hook to run when the event is executed
+		callable $callback, //the callback function of that hook
+		string $reccurence, //how often the event should subsequently recur
+		string $executionTime = '', //takes time of day the job should be executed, only for daily reccurence
+		array $option = array(), //first element is the options_key, second is the field_id. If set, the field is checked and determines wether the hook should be ran
+		string $updateHook= ''  //The wordpress hook that should update the option
+	)
+	{
+		// Add custom cron intervals
+		add_filter( 'cron_schedules', array( self::class, 'initIntervals' ) );
+
+		$jobhook = COMMONSBOOKING_PLUGIN_SLUG . '_' .$jobhook; //Prepends plugin slug so that hooks can be found easily afterwards 
```

Comment:
@hansmorb : Schreib das doch gleich in $this->jobhook und verwende das an den folgenden Stellen. Die Variable ist hier unnötig.